### PR TITLE
add metrics to cinder volume

### LINKED ISSUE #62644: How to get kubelet_volume* metrics working?
**Is this a BUG REPORT or FEATURE REQUEST?**: BUG

/kind bug

**What happened**: I am trying to get `kubelet_volume_*` metrics working. According this https://github.com/google/cadvisor/issues/1702#issuecomment-344324219 there should be support after 1.8 kube. However, I am now using kube 1.10.1 and still I cannot see anything in `/metrics`? Should these be there or not?

``` 
[root@kube-dev-node-1-1 log]# ifconfig eth0
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1450
        inet 192.168.18.9  netmask 255.255.255.0  broadcast 192.168.18.255
        inet6 fe80::f816:3eff:fe4f:a183  prefixlen 64  scopeid 0x20<link>
        ether fa:16:3e:4f:a1:83  txqueuelen 1000  (Ethernet)
        RX packets 9409  bytes 4087711 (3.8 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 8731  bytes 4873482 (4.6 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

[root@kube-dev-node-1-1 log]# curl -s https://192.168.18.9:10250/metrics -k|grep kubelet_volume|wc -L
0
[root@kube-dev-node-1-1 log]# df -h
Filesystem               Size  Used Avail Use% Mounted on
/dev/vda1                 11G  1.9G  8.1G  19% /
devtmpfs                 1.9G     0  1.9G   0% /dev
tmpfs                    1.9G     0  1.9G   0% /dev/shm
tmpfs                    1.9G  9.7M  1.9G   1% /run
tmpfs                    1.9G     0  1.9G   0% /sys/fs/cgroup
/dev/mapper/misc-docker   15G   47M   14G   1% /var/lib/docker
tmpfs                    1.9G   12K  1.9G   1% /var/lib/kubelet/pods/b79edabb-416a-11e8-9f3d-fa163e02c5d3/volumes/kubernetes.io~secret/default-token-hvf4r
tmpfs                    1.9G   12K  1.9G   1% /var/lib/kubelet/pods/c074683f-416a-11e8-9f3d-fa163e02c5d3/volumes/kubernetes.io~secret/weave-net-token-npm5t
tmpfs                    1.9G   12K  1.9G   1% /var/lib/kubelet/pods/b79dbbb3-416a-11e8-9f3d-fa163e02c5d3/volumes/kubernetes.io~secret/default-token-hvf4r
tmpfs                    1.9G   12K  1.9G   1% /var/lib/kubelet/pods/fbfe4b47-416a-11e8-9f3d-fa163e02c5d3/volumes/kubernetes.io~secret/default-token-jbbs6
/dev/vdb                 7.8G   36M  7.3G   1% /var/lib/kubelet/plugins/kubernetes.io/cinder/mounts/8810cc18-525b-42bb-a894-4c5eb1620305
tmpfs                    380M     0  380M   0% /run/user/1000
```

**What you expected to happen**: I except that kubelet should expose prometheus metrics to /metrics path as well. Then it is easy to read with prometheus.

**How to reproduce it (as minimally and precisely as possible)**: Create new cinder volume, attach it to pod and try to query result from `/metrics` in kubelet

**Environment**:
- Kubernetes version (use `kubectl version`): 1.10.1
- Cloud provider or hardware configuration: openstack
- OS (e.g. from /etc/os-release): centos 7
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by zetaab**
**What this PR does / why we need it**: We need cinder volume metrics to be exposed

**Which issue(s) this PR fixes**:
Fixes #62644

**Special notes for your reviewer**: This is not tested in running cluster. 

**Release note**:

```release-note
Make volume usage metrics available for Cinder
```

cc @gnufied @dims

### CODE REVIEW

**Comment by gnufied** [comment]
Just to double check, you have verified kubelet stats metrics after this commit right?

**Comment by zetaab** [comment]
nah, I have not verified it. I will check that tomorrow and confirm

File: pkg/volume/cinder/cinder.go
Comment (dims) [inline pkg/volume/cinder/cinder.go]: I am not comfortable with this change. Do they even point to the same directory?

**Comment by dims** [comment]
/hold

let's please not rush this in yet

File: pkg/volume/cinder/cinder.go
Comment (zetaab) [reply pkg/volume/cinder/cinder.go]: if you check that getPath function in rows 82-85 it will do the same thing. It is just common function for all of these getpath calls.

**Comment by zetaab** [comment]
with this code I can get following out of kubelet:

```
[root@kube-dev-node-1-1 log]# curl -s https://192.168.18.9:10250/metrics -k|grep kubelet_volume
# HELP kubelet_volume_stats_available_bytes Number of available bytes in the volume
# TYPE kubelet_volume_stats_available_bytes gauge
kubelet_volume_stats_available_bytes{namespace="default",persistentvolumeclaim="storage-myapp-0"} 7.836893184e+09
# HELP kubelet_volume_stats_capacity_bytes Capacity in bytes of the volume
# TYPE kubelet_volume_stats_capacity_bytes gauge
kubelet_volume_stats_capacity_bytes{namespace="default",persistentvolumeclaim="storage-myapp-0"} 8.32090112e+09
# HELP kubelet_volume_stats_inodes Maximum number of inodes in the volume
# TYPE kubelet_volume_stats_inodes gauge
kubelet_volume_stats_inodes{namespace="default",persistentvolumeclaim="storage-myapp-0"} 524288
# HELP kubelet_volume_stats_inodes_free Number of free inodes in the volume
# TYPE kubelet_volume_stats_inodes_free gauge
kubelet_volume_stats_inodes_free{namespace="default",persistentvolumeclaim="storage-myapp-0"} 524277
# HELP kubelet_volume_stats_inodes_used Number of used inodes in the volume
# TYPE kubelet_volume_stats_inodes_used gauge
kubelet_volume_stats_inodes_used{namespace="default",persistentvolumeclaim="storage-myapp-0"} 11
# HELP kubelet_volume_stats_used_bytes Number of used bytes in the volume
# TYPE kubelet_volume_stats_used_bytes gauge
kubelet_volume_stats_used_bytes{namespace="default",persistentvolumeclaim="storage-myapp-0"} 3.7736448e+07
```

File: pkg/volume/cinder/cinder.go
Comment (dims) [reply pkg/volume/cinder/cinder.go]: cool

**Comment by dims** [comment]
@zetaab let's squash this and give some time for folks to review

/assign @FengyunPan2 
/assign @anguslees

**Review by FengyunPan2 — APPROVED** [review]
/approve

**Comment by zetaab** [comment]
@gnufied added release-note

**Pull request merged**
